python/pyiceberg/table/__init__.py

```diff
@@ -183,14 +185,17 @@ class TableScan(Generic[S], ABC):
     def __init__(
         self,
         table: Table,
-        row_filter: Optional[BooleanExpression] = None,
+        row_filter: Optional[Union[str, BooleanExpression]] = None,
         selected_fields: Tuple[str] = ("*",),
         case_sensitive: bool = True,
         snapshot_id: Optional[int] = None,
         options: Properties = EMPTY_DICT,
     ):
         self.table = table
-        self.row_filter = row_filter or AlwaysTrue()
+        if row_filter is None:
+            self.row_filter = AlwaysTrue()
+        else:
```

Comment:
`elif isinstance(row_filter, str):`?

Reply:
We also do the same logic in the `filter()` method below, therefore I added it to the parse. But I agree, it will clutter up the parse method in the long run.